src/main/java/hu/dungeonhunter/tools/Fight.java

```diff
@@ -1,47 +1,100 @@
-        monster.setMonsterHP(monster.getMonsterHP() - champion.championDamage());
-        System.out.println("Monster have now " + monster.getMonsterHP() + " hit points");
-        if (monster.getMonsterHP() > 0)
-            monsterAttack();
-        else {
-            gameOver();
+    public Fight(){
+        monsterCounter = Dice.rollDice(6,2);
+        System.out.println(monsterCounter + " monsters is in the Dungeon!");
+        start();
+    }
+
+    public void start(){
+        if (monsterCounter != 0) {
+            System.out.println("Still left in the cave " + monsterCounter + " !");
+            monsterCaller();
+            actionMenu();
         }
+        else
+        System.out.println("The Dungeon is clear! You killed " + killedMonsterCounter + "monster (not counting the " +
+                "many mothers and children), you win!");
+    }
+
+    public Monsters monsterCaller() {
+        if (monsterCounter != 0) {
+            System.out.println("A monster step out from darkness!");
+            monster = new Monsters();
+        }return monster;
     }
 
     public void monsterAttack() {
         System.out.print("Monster attack: ");
-        champion.setChampionHP(champion.getChampionHP() - monster.monsterDamage());
-        System.out.println("Champion have now " + champion.getChampionHP() + " hit points");
-        gameOver();
+        champion.setHp(champion.getHp() - monster.monsterDamage());
+        System.out.println("Champion have now " + champion.getHp() + " hit points");
+        champion.enemyVictory();
+        if (!champion.isDefeat())
+            nextTurn();
     }
 
-    public void gameOver() {
-        if (champion.getChampionHP() > 0 && monster.getMonsterHP() < 1) {
-            System.out.println("You killed the monster! You win!");
-            champion.setChampionVictory(true);
+    public void championAttack() {
+        System.out.print("Champion attack: ");
+        monster.setHp(monster.getHp() - champion.championDamage());
+        System.out.println("Monster have now " + monster.getHp() + " hit points");
+        monster.enemyVictory();
+        if (!monster.isDefeat()){
+            monsterAttack();
         }
-        if (champion.getChampionHP() < 1 && monster.getMonsterHP() > 0) {
-            System.out.println("The monster killed you! You lost!");
-            monster.setMonsterVictory(true);
+        else{
+            killedMonsterCounter++;
+            monsterCounter--;
+            start();
         }
-        if (champion.getChampionHP() < 1 && monster.getMonsterHP() < 1)
-            System.out.println("you are both dead");
+    }
+
+    int chosenNumber() {
+        do {
+            Scanner scanner = new Scanner(System.in);
+            try {
+                return scanner.nextInt();
+            } catch (InputMismatchException hibafogo) {
+                yourChooseIsNotAppropriate();
+                printMenu();
+            }
+        } while (true);
```

Comment:
```suggestion
        do {
            printMenu();
            Scanner scanner = new Scanner(System.in);
            try {
                return scanner.nextInt();
            } catch (InputMismatchException hibafogo) {
                yourChooseIsNotAppropriate();
            }
        } while (true);
```
And remove printMenu(); from line 93

Reply:
Why? For dont take 2 line the printMenu method ?
I corrected.

Reply:
Yes, it is enough to call printMenu at one place. When the user is asked to enter an input, the menu will be automatically printed for the user.